features/nanostack/FEATURE_NANOSTACK/nanostack_entropy_check.cpp

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2015 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbedtls/platform.h"
+
+#if !defined(MBEDTLS_ENTROPY_HARDWARE_ALT) && !defined(MBEDTLS_TEST_NULL_ENTROPY)
```

Comment:
I don't think this is the right condition, because `MBEDTLS_ENTROPY_HARDWARE_ALT`is no longer defined in `target.json` and is a configuration of mbed TLS, not mbed OS.

Better to have
```
#if !defined(DEVICE_TRNG) && !defined(MBEDTLS_TEST_NULL_ENTROPY)
```

cc: @andresag01